distributed/client.py

```diff
@@ -394,6 +396,16 @@ def __init__(self, address=None, start=True, loop=None, timeout=5,
         self.connection_args = self.security.get_connection_args('client')
         self._connecting_to_scheduler = False
 
+        if loop is None:
+            self._should_close_loop = None
+            if in_ioloop():
+                self.loop = IOLoop.current()
```

Comment:
Oh... Well, I understand why: the logic is faulty. We're trying to get the loop for the current thread in `Client.__init__`, but then we call `Client.start()` which starts that loop in another thread. That's bound to fail in various ways depending on whether the loop is already used somewhere else.

I think we need a rethink of how Client chooses, initializes and starts its event loop.
Also, I think we should deprecate the explicit `loop` argument (not only in `Client.__init__`, but in other places too), because it seems both useless and fragile.